x-pack/plugins/enterprise_search/public/applications/app_search/components/engines/utils.ts

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { flatten } from 'lodash';
```

Comment:
Hey Constance, I've added some unit tests to utils.test.ts that should clarify why I'm flattening here (and I'm using `Array.prorotype.flatMap` now instead of flatten